connectors/src/connectors/zendesk/lib/data_cleanup.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  getArticleInternalId,
+  getTicketInternalId,
+} from "@connectors/connectors/zendesk/lib/id_conversions";
+import { deleteFromDataSource } from "@connectors/lib/data_sources";
+import {
+  ZendeskArticleResource,
+  ZendeskBrandResource,
+  ZendeskCategoryResource,
+  ZendeskTicketResource,
+} from "@connectors/resources/zendesk_resources";
+import type { DataSourceConfig } from "@connectors/types/data_source_config";
+
+/**
+ * Deletes all the data relative to a brand (tickets, help center, categories, articles) in the data source and in the db.
+ */
+export async function deleteBrand({
+  connectorId,
+  brandId,
+  dataSourceConfig,
+}: {
+  connectorId: number;
+  brandId: number;
+  dataSourceConfig: DataSourceConfig;
+}) {
+  await Promise.all([
+    deleteBrandHelpCenter({ connectorId, brandId, dataSourceConfig }),
+    deleteBrandTickets({ connectorId, brandId, dataSourceConfig }),
+  ]);
+  await ZendeskBrandResource.deleteByConnectorId(connectorId);
+}
+
+/**
+ * Deletes all the tickets stored in the db and in the data source relative to a brand.
+ */
+export async function deleteBrandTickets({
+  connectorId,
+  brandId,
+  dataSourceConfig,
+}: {
+  connectorId: number;
+  brandId: number;
+  dataSourceConfig: DataSourceConfig;
+}) {
+  const tickets = await ZendeskTicketResource.fetchByBrandId({
+    connectorId,
+    brandId,
+  });
+  /// deleting the tickets in the data source
+  await Promise.all(
```

Comment:
How many tickets can this be? Shouldn't this be in an activity by batch?

Reply:
It can be a lot, this should definitely be batched

Reply:
thanks for pointing out